bc_obps/compliance/service/compliance_dashboard_service.py

```diff
@@ -270,3 +272,48 @@ def _annotate_for_schema(
             ),
         )
         return cast(QuerySet[ComplianceReportVersion], annotated)
+
+    @staticmethod
+    def _annotate_status(
+        qs: QuerySet[ComplianceReportVersion],
+    ) -> QuerySet[ComplianceReportVersion]:
+        annotated = qs.annotate(
+            display_status=Case(
+                When(status__isnull=True, then=Value("N/A")),
+                When(status=ComplianceSummaryStatus.OBLIGATION_NOT_MET.value, then=Value("Obligation - not met")),
+                When(status=ComplianceSummaryStatus.OBLIGATION_FULLY_MET.value, then=Value("Obligation - met")),
+                When(
+                    status=ComplianceSummaryStatus.OBLIGATION_PENDING_INVOICE_CREATION.value,
+                    then=Value("Obligation - pending invoice creation"),
+                ),
+                When(
+                    status=ComplianceSummaryStatus.EARNED_CREDITS.value,
+                    issuance_status=IssuanceStatus.CREDITS_NOT_ISSUED.value,
+                    then=Value("Earned credits - not requested"),
+                ),
+                When(
+                    status=ComplianceSummaryStatus.EARNED_CREDITS.value,
+                    issuance_status=IssuanceStatus.ISSUANCE_REQUESTED.value,
+                    then=Value("Earned credits - issuance requested"),
+                ),
+                When(
+                    status=ComplianceSummaryStatus.EARNED_CREDITS.value,
+                    issuance_status=IssuanceStatus.APPROVED.value,
+                    then=Value("Earned credits - approved"),
+                ),
+                When(
+                    status=ComplianceSummaryStatus.EARNED_CREDITS.value,
+                    issuance_status=IssuanceStatus.DECLINED.value,
+                    then=Value("Earned credits - declined"),
+                ),
+                When(
+                    status=ComplianceSummaryStatus.EARNED_CREDITS.value,
+                    issuance_status=IssuanceStatus.CHANGES_REQUIRED.value,
+                    then=Value("Earned credits - changes required"),
+                ),
+                When(status=ComplianceSummaryStatus.EARNED_CREDITS.value, then=Value("Earned credits")),
+                default=F("status"),
+                output_field=CharField(),
+            ),
+        )
+        return cast(QuerySet[ComplianceReportVersion], annotated)
```

Comment:
Since we have some business logic here, I suggest adding a test to cover all the cases in this function, just to make sure we always end up with the correct status.